src/app/derive/derive.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import {AccountService} from '../account.service';
+import {IdentityService} from '../identity.service';
+import {BackendAPIService} from '../backend-api.service';
+import {GlobalVarsService} from '../global-vars.service';
+import {GoogleDriveService} from '../google-drive.service';
+
+@Component({
+  selector: 'app-derive',
+  templateUrl: './derive.component.html',
+  styleUrls: ['./derive.component.scss']
+})
+export class DeriveComponent implements OnInit {
+
+  allUsers: {[key: string]: any} = {};
+  hasUsers = false;
+  showAccessLevels = true;
+  withCallback = false;
+
+  constructor(
+    private accountService: AccountService,
+    private identityService: IdentityService,
+    public globalVars: GlobalVarsService,
+    private googleDrive: GoogleDriveService,
+    private backendApi: BackendAPIService,
+  ) { }
+
+  ngOnInit(): void {
+    // Load profile pictures and usernames
+    const publicKeys = this.accountService.getPublicKeys();
+    this.hasUsers = publicKeys.length > 0;
+    this.backendApi.GetUserProfiles(publicKeys)
+      .subscribe(profiles => {
+        this.allUsers = profiles;
+      });
+
+    // Set showAccessLevels
+    this.showAccessLevels = !this.globalVars.isFullAccessHostname();
+
+    // Set withCallback
+    this.withCallback = this.globalVars.callback !== null;
+  }
+
+  launchGoogle(): void {
+    this.googleDrive.launchGoogle();
+  }
+
+  selectAccount(publicKey: string): void {
```

Comment:
nit: I don't like that this has the same name as the log-in component. Can we rename this to selectAccountAndDeriveKey?